lib/src/drawing/sub_features/shape_drawing/painters/shape_painter.dart

```diff
@@ -74,25 +75,57 @@ base class ShapePainter extends DrawingPainter<ShapeDrawing> {
           element.operation == DrawingOperation.end,
     );
 
-    final double x1 = firstDelta.point.x;
-    final double x2 = secondDelta.point.x;
+    final PointDouble point1 = Point(firstDelta.point.x, firstDelta.point.y);
+    final PointDouble point2 = Point(secondDelta.point.x, secondDelta.point.y);
 
-    final double y1 = firstDelta.point.y;
-    final double y2 = secondDelta.point.y;
+    final PointDouble topVertex = PointDouble((point1.x + point2.x) / 2, point2.y);
```

Comment:
maybe this `(point1.x + point2.x) / 2` formula can be place in some utils package